codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -161,6 +161,17 @@ public void onHttpServerUpgrade(Http2Settings settings) throws Http2Exception {
         connection().remote().createStream(HTTP_UPGRADE_STREAM_ID, true);
     }
 
+    @Override
+    public void flush(ChannelHandlerContext ctx) throws Http2Exception {
+        // Trigger pending writes in the remote flow controller.
+        connection().remote().flowController().writePendingBytes();
```

Comment:
If we're coalescing we need to delay the writes, this is done by queueing them in the remote flow controller. We also need a point at which we don't delay writes any more and that is what is being done here right before flush.